File: menu/app/src/main/java/com/example/menu/MainActivity.java
package com.example.menu;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.main_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()){
            case R.id.item1:
                Toast.makeText(this, "You clicked on Settings", Toast.LENGTH_SHORT).show();
                return true;

            case R.id.item2:
                Toast.makeText(this, "You clicked on Search", Toast.LENGTH_SHORT).show();
                return true;

            case R.id.item3:
                Toast.makeText(this, "You clicked on Compose Email", Toast.LENGTH_SHORT).show();
                return true;

            case R.id.item4:
                Toast.makeText(this, "You clicked on Feedback", Toast.LENGTH_SHORT).show();
                return true;

            case R.id.sub_item1:
                Toast.makeText(this, "You clicked on Sub-Item 1", Toast.LENGTH_SHORT).show();
                return true;

            case R.id.sub_item2:
                Toast.makeText(this, "You clicked on Sub-Item 2", Toast.LENGTH_SHORT).show();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: BroadcastReceiver/app/src/main/java/com/example/broadcastreceiver/MainActivity.kt
package com.example.broadcastreceiver

import android.content.Context
import android.net.ConnectivityManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val c = applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = c.activeNetworkInfo

        if (networkInfo!= null && networkInfo.isConnected)
        {
            if (networkInfo.type==ConnectivityManager.TYPE_MOBILE)
            {
                Toast.makeText(applicationContext,"Connected to Mobile", Toast.LENGTH_LONG).show()
            }
            if (networkInfo.type==ConnectivityManager.TYPE_WIFI)
            {
                Toast.makeText(applicationContext, "Connected to Wifi", Toast.LENGTH_LONG).show()
            }
        }
        else
        {
            Toast.makeText(applicationContext, "You are Offline", Toast.LENGTH_LONG).show()
        }
    }
}